components/base/fake/fake.go

```diff
@@ -0,0 +1,173 @@
+// Package fake implements a fake base.
+package fake
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/components/base"
+	limo "go.viam.com/rdk/components/base/agilex"
+	"go.viam.com/rdk/components/base/boat"
+	"go.viam.com/rdk/components/base/wheeled"
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/services/slam"
+	"go.viam.com/rdk/spatialmath"
+)
+
+// ModelName is the name of the fake model of a base component.
+var ModelName = resource.NewDefaultModel("fake")
+
+// Config is used for converting config attributes.
+type Config struct {
+	BaseModel string `json:"base-model,omitempty"`
+}
+
+func init() {
+	registry.RegisterComponent(
+		base.Subtype,
+		ModelName,
+		registry.Component{
+			Constructor: func(
+				ctx context.Context,
+				_ registry.Dependencies,
+				config config.Component,
+				logger golog.Logger,
+			) (interface{}, error) {
+				return NewBase(ctx, config, logger)
+			},
+		},
+	)
+}
+
+var _ = base.LocalBase(&Base{})
+
+// Base is a fake base that returns what it was provided in each method.
+type Base struct {
+	generic.Echo
+	Name              string
+	CloseCount        int
+	logger            golog.Logger
+	modelName         string
+	collisionGeometry spatialmath.Geometry
+}
+
+// NewBase instantiates a new base of the fake model type.
+func NewBase(ctx context.Context, cfg config.Component, logger golog.Logger) (base.LocalBase, error) {
+	// TODO(rb): This should not ultimately be using the wheeled base package to do this
+	geometry, err := base.CollisionGeometry(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Base{
+		Name:              cfg.Name,
+		logger:            logger,
+		modelName:         cfg.ConvertedAttributes.(*Config).BaseModel,
+		collisionGeometry: geometry,
+	}, nil
+}
+
+// MoveStraight does nothing.
+func (b *Base) MoveStraight(ctx context.Context, distanceMm int, mmPerSec float64, extra map[string]interface{}) error {
+	return nil
+}
+
+// Spin does nothing.
+func (b *Base) Spin(ctx context.Context, angleDeg, degsPerSec float64, extra map[string]interface{}) error {
+	return nil
+}
+
+// SetPower does nothing.
+func (b *Base) SetPower(ctx context.Context, linear, angular r3.Vector, extra map[string]interface{}) error {
+	return nil
+}
+
+// SetVelocity does nothing.
+func (b *Base) SetVelocity(ctx context.Context, linear, angular r3.Vector, extra map[string]interface{}) error {
+	return nil
+}
+
+// Width returns some arbitrary width.
+func (b *Base) Width(ctx context.Context) (int, error) {
+	return 600, nil
```

Comment:
Would be neat if this was configurable.

Even if not configurable, should be a `const` up top rather than a magic number in the middle of the file.

Reply:
Yeah. I copy pasted this from the original file.  I will make this less bad

Reply:
Made it a constant.  Theres a larger project here that involves making this fully customizable but since Rand requested I hold off on some of those changes I'm going to do the simple thing here